script.js

```diff
@@ -1,4 +1,368 @@
+/* Note: This implementation incldues the following features:
+1. Regular beat that game (i.e. largest num wins)
+2. Score
+3. Leaderboard
+4. Lowest Combined Number Mode
+5. Auto-Generate Combined Number
+*/
+/*====================================
+==============GLOBAL VARIABLES========
+======================================*/
+// Game modes
+var GAME_MODE_DICE_ROLL = 'GAME_MODE_DICE_ROLL';
+var GAME_MODE_CHOOSE_DICE_ORDER_AUTOMATICALLY =
+  'GAME_MODE_CHOOSE_DICE_ORDER_AUTOMATICALLY';
+var REGULAR = 'regular';
+var LOWEST_COMBINED_NUMBER = 'lowest combined number';
+
+var mode = null;
+
+// Initialise the game to start with the dice roll game mode
+var gameMode = GAME_MODE_DICE_ROLL;
+
+// Keep track of the current player's number, either 1 or 2.
+// The game starts with Player 1.
+var currPlayer = 1;
+
+// Keep track of each player's dice rolls
+var player1Dice = [];
+var player2Dice = [];
+
+// Keep track of each player's chosen numbers
+var player1Num;
+var player2Num;
+
+// Use player profiles to keep track of each player's details
+var playerProfiles = [
+  {id: 1, score: 0},
+  {id: 2, score: 0},
+];
+
+/*====================================
+==============HELPER FUNCTIONS========
+======================================*/
```

Comment:
@jeremylim91 helper functions isn't a good header. all functions are helper functions except for `main`. could we either remove this or rename it?